### PR TITLE
Makes TGUI build also run prettier automatically

### PR DESCRIPTION
**Pull request by S34NW**
## What Does This PR Do
Changes TGUI build script to also run prettier automatically, so we don't have any out of compliance files.

Takes 2 seconds extra to run a TGUI build.

Adds some explicit requirements to avoid errors in certain situations.
Updates old configs to avoid errors.
Added windows recycle bin dir to gitignore, as some people will be using the root of Windows 11's new "dev drives" for perf optimization.

Runs prettier, so the next person who does a merge-master on TGUI PR or opens a TGUI PR won't get slapped with the changes.

## Why It's Good For The Game
We added the tool for a reason, lets make sure everyone runs it.

## Images of changes
image
this is running the build script from scratch

## Testing
Ran the updated build bat

### CODE REVIEW

File: tgui/packages/tgui/interfaces/NTRecruiter.js [hunk 1]
```diff
@@ -80,8 +80,9 @@ export const NTRecruiter = (props, context) => {
           >
             <LabeledList>
               <LabeledListItem label="1#" color="silver">
-                To win this game you must hire/dismiss <b>{total_curriculums}</b> candidates,
-                one wrongly made choice leads to a game over.
+                To win this game you must hire/dismiss{' '}
```
Comment (S34NW) [inline tgui/packages/tgui/interfaces/NTRecruiter.js:83]: this seems sus, @Sirryan2002 is this okay?

Comment (Henri215) [reply tgui/packages/tgui/interfaces/NTRecruiter.js:83]: Very sus

File: .gitignore [hunk 2]
```diff
@@ -58,3 +58,4 @@ __pycache__/
 dmm-tools.exe
 OpenDream
 paradise.json
+$RECYCLE.BIN
```
Comment (S34NW) [inline .gitignore:61]: I couldn't not add this as my dev environment is now within a windows dev drive root, won't cause any harm

File: tgui/bin/tgui-build.bat [hunk 3]
```diff
@@ -1,5 +1,6 @@
 @echo off
 cd "%~dp0\.."
 call yarn install
+call yarn prettier --write
 call yarn run build
-timeout /t 9
+pause
```
Comment (S34NW) [inline tgui/bin/tgui-build.bat:6]:
```suggestion
timeout /t 9
```

File: tgui/package.json [hunk 4]
```diff
@@ -15,6 +15,10 @@
   },
   "dependencies": {
     "@babel/eslint-parser": "^7.22.15",
+    "@babel/core": "^7.23.2",
+    "@babel/preset-env": "^7.23.2",
+    "@babel/plugin-proposal-class-properties": "^7.18.6",
+    "@babel/plugin-transform-jscript": "^7.22.5",
```
Comment (S34NW) [inline tgui/package.json:21]: These were required for the bat to run properly now, god knows why this crept in between me making the original prettier PR and now, probably some dependabot update screwed things

File: tgui/packages/tgui/interfaces/NTRecruiter.js (hunk 1 above)
Comment (S34NW) [reply tgui/packages/tgui/interfaces/NTRecruiter.js:83]: image

still works fine, so not sure why it is doing here

File: tgui/.eslintrc.yml [hunk 5]
```diff
@@ -1,4 +1,5 @@
 parser: '@babel/eslint-parser'
+root: ''
```
Comment (S34NW) [inline tgui/.eslintrc.yml:2]: helps eslint know where to look for configs

File: tgui/packages/tgui/interfaces/NTRecruiter.js (hunk 1 above)
Comment (Contrabang) [reply tgui/packages/tgui/interfaces/NTRecruiter.js:83]: Yes, this is fine

**Comment by S34NW** [comment]
Drafting while I figure out the stupid eslint parsing errors in vsc

```[{
	"resource": "/x:/tgui/packages/tgui/interfaces/NTRecruiter.js",
	"owner": "eslint",
	"severity": 4,
	"message": "Parsing error: No Babel config file detected for X:\\tgui\\packages\\tgui\\interfaces\\NTRecruiter.js. Either disable config file checking with requireConfigFile: false, or configure Babel so that it can find the config files.",
	"source": "eslint",
	"startLineNumber": 1,
	"startColumn": 1,
	"endLineNumber": 1,
	"endColumn": 1
}]```

**Comment by S34NW** [comment]
fixed

File: tgui/packages/tgui/interfaces/NTRecruiter.js (hunk 1 above)
Comment (Sirryan2002) [reply tgui/packages/tgui/interfaces/NTRecruiter.js:83]: its the same output, this is just something prettier does for some stupid reason, there's not a practical solution to it tbh

**Review by Sirryan2002 — APPROVED** [review]
everything looks good

**Review by Contrabang — APPROVED** [review]
LGTM

**Review by lewcc — APPROVED** [review]

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**